atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -246,8 +246,8 @@ private void dropTableInternal(String fullTableName, String shortTableName) {
     public void truncate() {
         try {
             conns.get()
-                    .executeUnregisteredQuery(
-                            "TRUNCATE TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef));
+                    .executeUnregisteredQuery("TRUNCATE TABLE "
+                            + oracleTableNameGetter.getInternalShortTableName(conns, tableRef) + " DROP STORAGE");
```

Comment:
@CVdV-au could we add a comment why `DROP STORAGE` and not `DROP ALL STORAGE`?

Reply:
Additionally, do we need a migration or manual cleanup to release storage & extents for previously truncated tables?

Reply:
DROP ALL STORAGE is just a more extreme version and releases whatever the MINEXTENT normally allocates. By default, this is 1 block, so not material. There is a non-trivial cost if all storage is deallocated when the table is used again. This is just a judgement without any science too it... 99.9% of the storage benefit comes from recovering the vast bulk of the allocated space.

We could add a migration. However, it seems very few consumers of this api are actually configured to use TRUNCATE - See PG-191739 for suggestion as to how we can change that around.